ci: Generate Duvet reports in CI

goatgoose opened the pull request:
### Resolved issues:

None

### Description of changes: 

Currently, [Duvet](https://github.com/aws/s2n-quic/tree/main/common/duvet) reports must be generated and shared manually for s2n-tls. This PR adds duvet initialization and report generation scripts, commits the generated spec files to compliance/specs, and adds a Github workflow to generate and publish Duvet reports on PR updates.

The workflow uses the following action, created for this PR:
~~https://github.com/goatgoose/duvet-action~~
https://github.com/aws/s2n-quic/pull/1370

### Call-outs:

None

### Testing:

 How is this change tested (unit tests, fuzz tests, etc.)? Are there any testing steps to be verified by the reviewer?

 Is this a refactor change? If so, how have you proved that the intended behavior hasn't changed?

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

camshaft commented on compliance/initialize_duvet.sh:
```diff
@@ -0,0 +1,7 @@
+#/usr/bin/env bash
+
+duvet extract https://tools.ietf.org/rfc/rfc5869 # HMAC-based Extract-and-Expand Key Derivation Function (HKDF)
```
One thing to note is `tools.ietf.org` now redirects to `www.rfc-editor.org`. It doesn't have to be a part of this PR but it would be good to use the canonical URL instead (see https://github.com/aws/s2n-quic/pull/1147).

goatgoose replied:
Okay, that makes sense. I think the easiest way to do this might be to wait until all the HRR PRs are merged and then fix this in a single PR, if that sounds reasonable. I created and assigned myself to #3377.

lrstewart commented on .github/workflows/ci_compliance.yml:
```diff
@@ -0,0 +1,21 @@
+    steps:
+      - uses: actions/checkout@v3
+      - uses: goatgoose/duvet-action@v1.0.0
+        with:
+          report-script: compliance/generate_report.sh
```
When we talked about this, you mentioned not committing all the spec files and just downloading them as part of the action. Why commit them?

camshaft replied:
The nice thing about committing them is the check is completely offline. This, for example, removes the requirement of the IETF site to be available.

goatgoose replied:
Originally I wasn't really sure which to do. I liked the idea of not needing to commit all of the spec files, but also liked that extra steps wouldn't be needed to generate the report locally, since I was planning on having the RFCs be an input to the action. However, thinking about it now, I probably could just include that part in the script passed to the action, so it would generate them locally also. So maybe this is an option? The offline aspect of committing them does makes sense though.

lrstewart commented on .github/workflows/ci_compliance.yml:
```diff
@@ -0,0 +1,21 @@
+  duvet:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: goatgoose/duvet-action@v1.0.0
```
Should we be using an action in your private repo?

goatgoose replied:
Moved to the action in s2n-quic in 447bc32

goatgoose commented:
To move off of my personal repo, I added the action in s2n-quic: https://github.com/aws/s2n-quic/pull/1370. We can run this action in s2n-tls by cloning s2n-quic and just pathing to the Duvet action:
https://github.com/aws/s2n-tls/blob/8487485aa0b73c2c37ef99d674bb29b424a603c4/.github/workflows/ci_compliance.yml#L14-L19

I tested the new action in s2n-quic by setting the ref in actions/checkout to my branch: 447bc32

After confirming the action works, I removed this in 8487485 so that it pulls from main. So, the check will fail until https://github.com/aws/s2n-quic/pull/1370 is merged.
